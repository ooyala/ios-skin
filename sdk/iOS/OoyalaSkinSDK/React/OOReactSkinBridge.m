//
//  OOReactSkinBridge.m
//  OoyalaSkinSDK
//
//  Created on 8/14/18.
//  Copyright Â© 2018 ooyala. All rights reserved.
//

#import "OOReactSkinBridge.h"
#import "OOReactSkinBridgeModule.h"
#import <React/RCTBridge+Private.h>

@interface OOReactSkinBridge ()

@property (nonatomic, weak, readonly) id<OOReactSkinBridgeDelegate> delegate;

@end


@implementation OOReactSkinBridge

@dynamic delegate;

- (instancetype)initWithDelegate:(id<OOReactSkinBridgeDelegate>)delegate
                   launchOptions:(NSDictionary *)launchOptions {
  return [super initWithDelegate:delegate launchOptions:launchOptions];
}

- (void)invalidate {
  [super invalidate];
  [NSNotificationCenter.defaultCenter removeObserver:self];
}

- (void)setUp {
  [super setUp];
  [NSNotificationCenter.defaultCenter addObserver:self
                                         selector:@selector(didInitializeModuleNotification:)
                                             name:RCTDidInitializeModuleNotification
                                           object:self.batchedBridge];
}

- (void)didInitializeModuleNotification:(NSNotification *)notification {
  id module = notification.userInfo[@"module"];

  if ([self.delegate respondsToSelector:@selector(bridge:didLoadModule:)]) {
    if (module && [module conformsToProtocol:@protocol(OOReactSkinBridgeModule)]) {
      [self.delegate bridge:self didLoadModule:module];
    }
  }
}

@end


@implementation RCTBridge (OOReactSkinEventsEmitterable)

- (OOReactSkinEventsEmitter *)skinEventsEmitter {
  return [self moduleForName:[OOReactSkinEventsEmitter moduleName]];
}

@end
