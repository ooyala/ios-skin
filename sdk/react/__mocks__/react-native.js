var ReactNative = {};

ReactNative.StyleSheet = {
	create: function(styles) {
		return styles;
	}
};

module.exports = ReactNative;