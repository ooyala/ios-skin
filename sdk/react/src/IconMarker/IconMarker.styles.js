// @flow

import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  root: {
    bottom: 10,
    height: 40,
    left: 0,
    position: 'absolute',
    width: 40,
  },
});
