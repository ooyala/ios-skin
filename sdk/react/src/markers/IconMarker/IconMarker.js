// @flow

import * as React from 'react';
import { Image, TouchableWithoutFeedback } from 'react-native';
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';

import styles from './IconMarker.styles';

type Props = {
  iconUrl: ?string,
  imageUrl: ?string,
  onSeek: () => void,
  style?: ViewStyleProp,
};

type State = {
  isExpanded: boolean,
};

export default class IconMarker extends React.Component<Props, State> {
  static defaultProps = {
    style: undefined,
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
    };

    (this: Object).handlePress = this.handlePress.bind(this);
  }

  handlePress() {
    const { imageUrl, onSeek } = this.props;
    const { isExpanded } = this.state;

    // Trigger seek callback if the marker has no image or has been expanded. If there is an image we have to show it
    // first and only after the second click on that image trigger the callback.
    if (!imageUrl || isExpanded) {
      onSeek();
    }

    this.setState({
      isExpanded: !isExpanded,
    });
  }

  render() {
    const { iconUrl, imageUrl, style } = this.props;
    const { isExpanded } = this.state;

    // If the marker is expanded and there is `imageUrl`, we have to use it.
    const image = (isExpanded && imageUrl ? imageUrl : iconUrl);

    // Image can still be absent.
    if (!image) {
      return null;
    }

    return (
      <TouchableWithoutFeedback onPress={this.handlePress}>
        <Image source={{ uri: image }} style={[styles.root, style]} />
      </TouchableWithoutFeedback>
    );
  }
}
