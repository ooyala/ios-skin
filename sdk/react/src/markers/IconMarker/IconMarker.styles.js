// @flow

import { StyleSheet } from 'react-native';

import { MARKERS_SIZES } from '../../../constants';
import commonMarkerStyle from '../styles/commonMarkerStyle';
import triangleStyle from '../styles/triangleStyle';

export default StyleSheet.create({
  expanded: {
    height: MARKERS_SIZES.ICON_EXPANDED_SIZE,
    transform: [
      { translateX: MARKERS_SIZES.ICON_EXPANDED_SIZE / -2 },
    ],
    width: MARKERS_SIZES.ICON_EXPANDED_SIZE,
  },
  image: {
    height: '100%',
    width: '100%',
  },
  root: {
    ...commonMarkerStyle,
    height: MARKERS_SIZES.ICON_COLLAPSED_SIZE,
    transform: [
      { translateX: MARKERS_SIZES.ICON_COLLAPSED_SIZE / -2 },
    ],
    width: MARKERS_SIZES.ICON_COLLAPSED_SIZE,
  },
  triangle: triangleStyle,
});
