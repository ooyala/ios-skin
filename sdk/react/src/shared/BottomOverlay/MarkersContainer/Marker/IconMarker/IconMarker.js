// @flow

import React from 'react';
import { Image, TouchableWithoutFeedback, View } from 'react-native';
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';

import createStyles from './IconMarker.styles';

type Props = {
  backgroundColor?: string,
  containerWidth: number,
  iconUrl?: string,
  imageUrl?: string,
  leftPosition: number,
  onSeek: () => void,
  onTouch: () => void,
  style?: ViewStyleProp,
  touchColor?: string,
};

type State = {
  isExpanded: boolean,
};

export default class IconMarker extends React.Component<Props, State> {
  static defaultProps = {
    backgroundColor: undefined,
    iconUrl: undefined,
    imageUrl: undefined,
    style: undefined,
    touchColor: undefined,
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
    };

    (this: Object).handlePress = this.handlePress.bind(this);
  }

  handlePress() {
    this.setState(({ isExpanded }, { imageUrl, onSeek, onTouch }) => {
      onTouch();

      // Trigger seek callback if the marker has no image or has been expanded. If there is an image we have to show it
      // first and only after the second click on that image trigger the callback.
      if (!imageUrl || isExpanded) {
        onSeek();
      }

      return {
        isExpanded: !isExpanded,
      };
    });
  }

  render() {
    const {
      backgroundColor, containerWidth, iconUrl, imageUrl, leftPosition, style, touchColor,
    } = this.props;
    const { isExpanded } = this.state;

    const styles = createStyles(leftPosition, containerWidth);

    // If the marker is expanded and there is `imageUrl`, we have to use it.
    const image = (isExpanded && imageUrl ? imageUrl : iconUrl);

    // Image can still be absent.
    if (!image) {
      return null;
    }

    const expandedStyles = [];

    // Apply expanded style only if the larger image is present.
    if (isExpanded && imageUrl) {
      expandedStyles.push(styles.expanded);

      if (touchColor) {
        expandedStyles.push({ backgroundColor: touchColor });
      }
    }

    return (
      <TouchableWithoutFeedback onPress={this.handlePress}>
        <View
          style={[
            styles.root,
            style,
            backgroundColor && { backgroundColor },
            ...expandedStyles,
          ]}
        >
          <Image source={{ uri: image }} style={styles.image} />
          <View
            style={[
              styles.triangle,
              // Apply expanded style only if the larger image is present.
              isExpanded && imageUrl && touchColor && { borderTopColor: touchColor },
              isExpanded && imageUrl && styles.triangleExpanded,
            ]}
          />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}
