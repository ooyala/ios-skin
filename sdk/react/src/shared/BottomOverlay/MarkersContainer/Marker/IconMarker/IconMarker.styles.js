// @flow

import { StyleSheet } from 'react-native';

import { MARKERS_SIZES } from '../../../../../constants';
import commonMarkerStyle from '../styles/commonMarker.style';
import triangleStyle from '../styles/triangle.style';

export default (leftPosition: number, containerWidth: number) => StyleSheet.create({
  expanded: {
    height: MARKERS_SIZES.ICON_EXPANDED_SIZE,
    transform: [
      { translateX: MARKERS_SIZES.ICON_EXPANDED_SIZE / -2 },
    ],
    width: MARKERS_SIZES.ICON_EXPANDED_SIZE,
  },
  image: {
    height: '100%',
    width: '100%',
  },
  root: {
    ...commonMarkerStyle,
    height: MARKERS_SIZES.ICON_COLLAPSED_SIZE,
    left: leftPosition,
    transform: [
      { translateX: MARKERS_SIZES.ICON_COLLAPSED_SIZE / -2 },
    ],
    width: MARKERS_SIZES.ICON_COLLAPSED_SIZE,
  },
  triangle: triangleStyle,
});
