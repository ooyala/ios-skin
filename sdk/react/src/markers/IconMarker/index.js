// @flow

import IconMarker from './IconMarker';

export default IconMarker;
