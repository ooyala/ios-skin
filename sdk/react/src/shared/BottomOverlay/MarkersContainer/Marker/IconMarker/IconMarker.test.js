// @flow

import { shallow } from 'enzyme';
import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';

import IconMarker from './IconMarker';

describe('IconMarker', () => {
  it('renders matching snapshot', () => {
    const wrapper = shallow(
      <IconMarker
        iconUrl="http://example.com/icon.png"
        imageUrl="http://example.com/image.png"
        onSeek={() => undefined}
      />,
    );

    expect(wrapper)
      .toMatchSnapshot();
  });

  it('does not render anything when URLs are empty', () => {
    const wrapper = shallow(
      <IconMarker
        iconUrl={undefined}
        imageUrl={undefined}
        onSeek={() => undefined}
      />,
    );

    expect(wrapper)
      .toEqual({});
  });

  it('triggers `onSeek` callback every time when pressed if `imageUrl` is empty', () => {
    const onSeekMock = jest.fn();
    const wrapper = shallow(
      <IconMarker
        iconUrl="http://example.com/icon.png"
        imageUrl={undefined}
        onSeek={onSeekMock}
      />,
    );

    wrapper.find(TouchableWithoutFeedback)
      .simulate('press');

    expect(onSeekMock)
      .toBeCalled();
  });

  it('triggers `onSeek` callback when pressed being expanded', () => {
    const onSeekMock = jest.fn();
    const wrapper = shallow(
      <IconMarker
        iconUrl="http://example.com/icon.png"
        imageUrl="http://example.com/image.png"
        onSeek={onSeekMock}
      />,
    );

    wrapper.find(TouchableWithoutFeedback)
      .simulate('press');

    expect(onSeekMock)
      .not
      .toBeCalled();

    wrapper.find(TouchableWithoutFeedback)
      .simulate('press');

    expect(onSeekMock)
      .toBeCalled();
  });
});
