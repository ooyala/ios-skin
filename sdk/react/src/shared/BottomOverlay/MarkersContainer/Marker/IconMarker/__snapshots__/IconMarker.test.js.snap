// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconMarker renders matching snapshot 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "borderColor": "#4E4E4E",
          "borderRadius": 2,
          "borderWidth": 1,
          "bottom": 6,
          "height": 32,
          "left": 0,
          "padding": 2,
          "position": "absolute",
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 4,
          "transform": Array [
            Object {
              "translateX": -16,
            },
          ],
          "width": 32,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Component
      source={
        Object {
          "uri": "http://example.com/icon.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 0,
            "borderColor": "transparent",
            "borderTopColor": "#4E4E4E",
            "borderWidth": 6,
            "bottom": -6,
            "height": 0,
            "left": "50%",
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": -3,
              },
            ],
            "width": 0,
          },
          false,
        ]
      }
    />
  </View>
</TouchableWithoutFeedback>
`;
